### ISSUE TITLE
CMake script ZeekPluginDynamic.cmake broken due to removal of FindCAF.cmake

### ISSUE THREAD

**Issue by bbannier** [zeek/zeek#1347]
zeek/cmake@eeae54e7678706d012b09e31a3ef046dad8ef183 removed the script `cmake/FindCAF.cmake`. The still existing script `cmake/ZeekPluginDynamic.cmake` on the other hand still needs to be able to find the CAF installation since it calls

```
find_package(CAF COMPONENTS core io openssl REQUIRED)
```

Since the CAF package is not in the search paths CMake tries to use a `FindCAF.cmake` in its module search path to locate CAF.

```
CMake Error at /opt/zeek-nightly/share/zeek/cmake/ZeekPluginDynamic.cmake:167 (find_package):
  By not providing "FindCAF.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "CAF", but
  CMake did not find one.
  Could not find a package configuration file provided by "CAF" with any of
  the following names:
    CAFConfig.cmake
    caf-config.cmake
  Add the installation prefix of "CAF" to CMAKE_PREFIX_PATH or set "CAF_DIR"
  to a directory containing one of the above files.  If "CAF" provides a
  separate development package or SDK, be sure it has been installed.
```

We probably need to pass a CAF location to `find_package` so it can locate the CMake package. If CAF was installed together via Zeek (like e.g., when installing from distro packages), `BRO_CONFIG_PREFIX` might be a good starting point for defining a CAF location.

**Comment by jsiwek** [comment]
Thanks, I was able to reproduce it by installing CAF separately, then using build/install Zeek with `--with-caf=`, then configure a plugin against that Zeek install.  Patch coming via https://github.com/zeek/cmake/pull/28 if you'd like to test.

**Comment by bbannier** [comment]
Thanks for looking into this so quickly @jsiwek. I left a :shipit: on zeek/cmake#28 since it does indeed fix the issue.

Since 4.0.0 is still in the rc stages, is there any reason to not include that patch there as well?

**Comment by jsiwek** [comment]
> Since 4.0.0 is still in the rc stages, is there any reason to not include this patch there as well instead of just a 4.1.0 target?

It could be, but also `FindCAF.cmake` is still shipped with 4.0.0, so there's no real bug or problem there AFAICT and prefer to not pull in changes that don't fix/improve anything -- I just did the same end-to-end test against 4.0.0 and didn't reproduce the problem, but let me know if you find otherwise.

**Comment by bbannier** [comment]
Ah I see, thanks for checking. I wasn't aware that bump of zeek/cmake happened after 4.0.0 was branched off.
